flink-runtime/src/test/java/org/apache/flink/runtime/state/StateBackendTestBase.java

```diff
@@ -1080,23 +1088,41 @@ public void testKryoRegisteringRestoreResilienceWithRegisteredSerializer() throw
 
             // on the second restore, since the custom serializer will be used for
             // deserialization, we expect the deliberate failure to be thrown
-            expectedException.expect(
-                    anyOf(
-                            isA(ExpectedKryoTestException.class),
-                            Matchers.<Throwable>hasProperty(
-                                    "cause", isA(ExpectedKryoTestException.class))));
-
-            // state backends that eagerly deserializes (such as the memory state backend) will fail
-            // here
-            backend = restoreKeyedBackend(IntSerializer.INSTANCE, snapshot2, env);
-
-            state =
-                    backend.getPartitionedState(
-                            VoidNamespace.INSTANCE, VoidNamespaceSerializer.INSTANCE, kvId);
-
-            backend.setCurrentKey(1);
-            // state backends that lazily deserializes (such as RocksDB) will fail here
-            state.value();
+            assertThatThrownBy(
+                            () -> {
+                                // state backends that eagerly deserializes (such as the memory
+                                // state backend) will fail
+                                // here
+                                CheckpointableKeyedStateBackend<Integer> restoreBackend = null;
+                                try {
+                                    restoreBackend =
+                                            restoreKeyedBackend(
+                                                    IntSerializer.INSTANCE, snapshot2, env);
+
+                                    ValueState<TestPojo> restoreState =
+                                            restoreBackend.getPartitionedState(
+                                                    VoidNamespace.INSTANCE,
+                                                    VoidNamespaceSerializer.INSTANCE,
+                                                    new ValueStateDescriptor<>("id", pojoType));
+
+                                    restoreBackend.setCurrentKey(1);
+                                    // state backends that lazily deserializes (such as RocksDB)
+                                    // will fail here
+                                    restoreState.value();
+                                } finally {
+                                    if (restoreBackend != null) {
+                                        restoreBackend.dispose();
+                                    }
+                                }
+                            })
+                    .satisfiesAnyOf(
+                            actual ->
+                                    assertThat(actual)
+                                            .isInstanceOf(ExpectedKryoTestException.class),
+                            actual ->
+                                    assertThat(actual)
+                                            .hasFieldOrProperty("cause")
```

Comment:
Could you please clarify why should we check it?
In my opinion `Throwable` has `cause` field and this check will always pass for anything extending `Throwable`

Reply:
The same condition as mentioned in the former question.